pkg/yamlmeta/type.go

```diff
@@ -103,18 +105,48 @@ func (t *MapType) AssignTypeTo(typeable Typeable) (chk TypeCheck) {
 		chk.Violations = []string{fmt.Sprintf("Expected node at %s to be a %T, but was a %T", typeable.GetPosition().AsCompactString(), &Map{}, typeable)}
 		return
 	}
+	var foundKeys []interface{}
 	typeable.SetType(t)
 	for _, mapItem := range mapNode.Items {
 		for _, itemType := range t.Items {
 			if mapItem.Key == itemType.Key {
+				foundKeys = append(foundKeys, itemType.Key)
 				childCheck := itemType.AssignTypeTo(mapItem)
 				chk.Violations = append(chk.Violations, childCheck.Violations...)
 				break
 			}
 		}
 	}
+
+	t.applySchemaDefaults(foundKeys, chk, mapNode)
 	return
 }
+
+func (t *MapType) applySchemaDefaults(foundKeys []interface{}, chk TypeCheck, mapNode *Map) {
+	for _, item := range t.Items {
+		if contains(foundKeys, item.Key) {
+			continue
+		}
+
+		val := &MapItem{
+			Key:      item.Key,
+			Value:    item.DefaultValue,
+			Position: item.Position,
+		}
+		childCheck := item.AssignTypeTo(val)
+		chk.Violations = append(chk.Violations, childCheck.Violations...)
+		mapNode.AddValue(val) // TODO: How might we surface this error?
```

Comment:
Open `TODO:`.

I think this is an unexpected situation. Are there cases where applying the default value _could_ result in a **type-check** violation (as opposed to a validation violation)?

Given that considering, what should we do, here?